File: tools_box/tools_box/report/computing_asset_inspection_checklist_report/computing_asset_inspection_checklist_report.py
# Copyright (c) 2013, bobzz.zone@gmail.com and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
#Employee	Computing Asst Type	Status	Remarks	##Employee Sign
def execute(filters=None):
	columns, data = ["Employee::200","Computing Asset Type::200","Status::150","Remark::200"], []
	status=""
	if filters.get("status"):
		status=""" and d.status = "{}" """.format(filters.get("status"))
	data = frappe.db.sql("""select e.employee_name,d.computing_asset_type,d.status,d.remark 
		from `tabComputing Asset Inspection Item` d 
		join `tabComputing Asset Inspection Checklist` p on d.parent=p.name 
		join `tabEmployee` e on d.employee = e.name
		where p.docstatus=1 and (p.date between "{}" and "{}") {} """.format(filters.get("from"),filters.get("to"),status),as_list=1)
	return columns, data

File: tools_box/tools_box/report/computing_asset_inspection_checklist_report/computing_asset_inspection_checklist_report.js
// Copyright (c) 2016, bobzz.zone@gmail.com and contributors
// For license information, please see license.txt

frappe.query_reports["Computing Asset Inspection Checklist Report"] = {
	"filters": [
		{
			"fieldname":"from",
			"label": __("From Date"),
			"fieldtype": "Date",
			"width": "80",
			"reqd":1
		},
		{
			"fieldname":"to",
			"label": __("To Date"),
			"fieldtype": "Date",
			"default": get_today(),
			"reqd":1
		},
		{
			"fieldname":"status",
			"label": __("Status"),
			"fieldtype": "Link",
			"options": "Computing Asset Inspection Status"
		},
	]
}
